### PR TITLE
Gtest

### PR DESCRIPTION
**Pull request by bitmouse**

### CODE REVIEW

File: tests/unittest/CMakeLists.txt [hunk 1]
```diff
@@ -0,0 +1,16 @@
+
+set(FACELIFT_GTEST_LIBS gtest gmock gtest_main)
+
+add_executable(runUnitTests
```
Comment (jacky309) [inline tests/unittest/CMakeLists.txt]: use  facelift_add_test()

Comment (bitmouse) [reply tests/unittest/CMakeLists.txt]: To be fixed.

File: .travis.yml [hunk 2]
```diff
@@ -48,10 +48,10 @@ before_script:
 - source /opt/qt${QT}/bin/qt${QT}-env.sh
 - cd ${TRAVIS_BUILD_DIR}
 - mkdir build
+- source setup_gtest.sh
```
Comment (jacky309) [inline .travis.yml]: provide a fixed SHA or tag name

File: CMakeLists.txt [hunk 3]
```diff
@@ -24,6 +24,7 @@ option(FACELIFT_BUILD_EXAMPLES "Enable build of examples" ON)
 option(FACELIFT_BUILD_TESTS "Enable build of tests" OFF)
 option(FACELIFT_ENABLE_DESKTOP_DEV_TOOLS "Enable desktop development tools" OFF)
 option(BUILD_SHARED_LIBS "Build shared libraries. If set to off, static libraries are built instead" ON)
+option(FACELIFT_DISABLE_GTEST "Use this flag to prevent build unit tests based on googletest" OFF)
```
Comment (jacky309) [inline CMakeLists.txt]: "Disable GTest-based unit tests"

File: tests/unittest/CMakeLists.txt [hunk 4]
```diff
@@ -0,0 +1,21 @@
...
+        set_target_properties(GTest::GMock PROPERTIES IMPORTED_LOCATION ${GMOCK_LIBRARY})
+    else()
+        message(WARNING "Google test/mock not found.")
+    endif()
+
+    find_package(Threads REQUIRED)
+    set(FACELIFT_GTEST_LIBRARIES ${GTEST_BOTH_LIBRARIES} GTest::GMock Threads::Threads)
+    include_directories(${GTEST_INCLUDE_DIRS})
+
+    include(testList.cmake)
```
Comment (jacky309) [inline tests/unittest/CMakeLists.txt]: move the content of that file in here.

**Pull request merged**